pyaedt/modeler/cad/Modeler.py

```diff
@@ -2653,36 +2634,63 @@ def mirror(self, objid, position, vector):
         position : int or float
             List of the ``[x, y, z]`` coordinates or the
             Application.Position object for the selection.
+        duplicate : bool, optional
+            Whether if duplicate the object before mirror or not. Default is ``False``.
+        is_3d_comp : bool, optional
+            If ``True``, the method will try to return the duplicated list of 3dcomponents. The default is ``False``.
         vector : float
             List of the ``[x1, y1, z1]`` coordinates or
             the Application.Position object for the vector.
+        duplicate_assignment : bool, optional
+            If True, the method duplicates selection assignments. The default value is ``True``.
 
         Returns
         -------
-        bool
-            ``True`` when successful, ``False`` when failed.
+        bool, list
+            List of objects created or ``True`` when successful, ``False`` when failed.
 
         References
         ----------
 
         >>> oEditor.Mirror
+        >>> oEditor.DuplicateMirror
         """
+
         selections = self.convert_to_selections(objid)
         Xpos, Ypos, Zpos = self._pos_with_arg(position)
         Xnorm, Ynorm, Znorm = self._pos_with_arg(vector)
-
-        vArg1 = ["NAME:Selections", "Selections:=", selections, "NewPartsModelFlag:=", "Model"]
-        vArg2 = ["NAME:MirrorParameters"]
-        vArg2.append("MirrorBaseX:="), vArg2.append(Xpos)
-        vArg2.append("MirrorBaseY:="), vArg2.append(Ypos)
-        vArg2.append("MirrorBaseZ:="), vArg2.append(Zpos)
-        vArg2.append("MirrorNormalX:="), vArg2.append(Xnorm)
-        vArg2.append("MirrorNormalY:="), vArg2.append(Ynorm)
-        vArg2.append("MirrorNormalZ:="), vArg2.append(Znorm)
-
-        self.oeditor.Mirror(vArg1, vArg2)
-
-        return True
+        if duplicate:
+            vArg1 = ["NAME:Selections", "Selections:=", selections, "NewPartsModelFlag:=", "Model"]
+            vArg2 = ["NAME:DuplicateToMirrorParameters"]
+            vArg2.append("DuplicateMirrorBaseX:="), vArg2.append(Xpos)
+            vArg2.append("DuplicateMirrorBaseY:="), vArg2.append(Ypos)
+            vArg2.append("DuplicateMirrorBaseZ:="), vArg2.append(Zpos)
+            vArg2.append("DuplicateMirrorNormalX:="), vArg2.append(Xnorm)
+            vArg2.append("DuplicateMirrorNormalY:="), vArg2.append(Ynorm)
+            vArg2.append("DuplicateMirrorNormalZ:="), vArg2.append(Znorm)
+            vArg3 = ["NAME:Options", "DuplicateAssignments:=", duplicate_assignment]
+            if is_3d_comp:
+                orig_3d = [i for i in self.user_defined_component_names]
+            added_objs = _retry_ntimes(10, self.oeditor.DuplicateMirror, vArg1, vArg2, vArg3)
+            self.add_new_objects()
+            if is_3d_comp:
+                added_3d_comps = [i for i in self.user_defined_component_names if i not in orig_3d]
+                if added_3d_comps:
+                    self.logger.info("Found 3D Components Duplication")
```

Comment:
Message should be written as a complete sentence, perhaps something like "3D components were duplicated."